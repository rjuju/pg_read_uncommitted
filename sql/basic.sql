CREATE TABLE read_uncommitted(id integer, val text);

INSERT INTO read_uncommitted SELECT 1, NULL;
UPDATE read_uncommitted SET val = 'first update';
UPDATE read_uncommitted SET id = 2, val = 'second update';
DELETE FROM read_uncommitted;

SELECT * FROM read_uncommitted ORDER BY id, val;

LOAD 'pg_read_uncommitted';
SET pg_read_uncommitted.enabled = true;
SET pg_read_uncommitted.show_deleted = true;

SELECT * FROM read_uncommitted ORDER BY id, val;
