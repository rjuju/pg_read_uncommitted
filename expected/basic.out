CREATE TABLE read_uncommitted(id integer, val text);
INSERT INTO read_uncommitted SELECT 1, NULL;
UPDATE read_uncommitted SET val = 'first update';
UPDATE read_uncommitted SET id = 2, val = 'second update';
DELETE FROM read_uncommitted;
SELECT * FROM read_uncommitted ORDER BY id, val;
 id | val 
----+-----
(0 rows)

LOAD 'pg_read_uncommitted';
SET pg_read_uncommitted.enabled = true;
SET pg_read_uncommitted.show_deleted = true;
SELECT * FROM read_uncommitted ORDER BY id, val;
 id |      val      
----+---------------
  1 | first update
  1 | 
  2 | second update
(3 rows)

